src/pages/index.js

```diff
@@ -0,0 +1,311 @@
+  "#profile__edit-modal",
+  handleProfileModalSubmit
+);
+profilePopup.setEventListeners();
+
+//Instantiate popupWithForm for addCard
+const newCardPopup = new PopupWithForm(
+  "#card__add-modal",
+  handleAddCardFormSubmit
+);
+newCardPopup.setEventListeners();
+//////Preview Image popup class
+const popupImage = new PopupWithImage("#preview__modal");
+popupImage.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#avatar__modal", handleAvatarUpdate);
+avatarPopup.setEventListeners();
+
+const deleteConfirm = new PopupWithConfirm("#delete__modal");
+deleteConfirm.setEventListeners();
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4792ec92-cf1c-45a6-8740-0f5d63585faa",
+    "Content-type": "application/json",
+  },
+});
+
+//////////////////////////////////////////// FUNCTIONS //////////////////////////////////////////////////////
+
+function handleImageClick(name, link) {
+  popupImage.openPopup({ name, link });
+}
+
+function handleProfileModalSubmit(profileData) {
+  profileModalSubmitButton.textContent = "Saving...";
+  console.log("profileData:", profileData);
+  const name = profileData.title; //set name var to work with userInfo class
+  const job = profileData.description; //set job var to work with userInfo class
+  console.log("name:", name, "job:", job); //debug log
+  user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
+  const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
+  console.log(updatedUserInfo); // debug log
+  profilePopup.closePopup(); //close popup
+
+  setTimeout(() => {
+    profileModalSubmitButton.textContent = "Save";
+  }, 1000);
+}
+
+function handleAvatarUpdate(link) {
+  const url = link.url; //extract url from link object and set it to url value
+  console.log(url);
+
+  avatarModalSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfilePic(url)
+    .then(() => {
+      user.setUserAvatar(url);
+    })
+    .catch((err) => {
+      console.error(`Error caught in HandleAvatarUpdate function ${err}`);
+    })
+    .finally(() => {
+      setTimeout(() => {
+        avatarModalSubmitButton.textContent = "Save";
+      }, 1000);
+    });
+  avatarPopup.closePopup();
+}
+
+function handleAddCardFormSubmit(newCardData) {
+  addNewCardSubmitButton.textContent = "Adding...";
+  api
+    .createNewCard(newCardData.title, newCardData.url)
+    .then((res) => {
+      /* console.log(res); */
+      renderCard(res);
+      newCardPopup.closePopup(); //close popup
+      addNewCardValidator.resetForm(); //reset form and disable submit button
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setTimeout(() => {
+        addNewCardSubmitButton.textContent = "Add";
+      }, 1000);
+    });
+}
+
+function handleCardLikes(cardData) {
+  if (!cardData._like) {
```

Comment:
Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.